experimenter/experimenter/experiments/models.py

```diff
@@ -369,26 +369,26 @@ def targeting(self):
             sticky_expressions.append(f"region in {countries}")
 
         if self.is_sticky and sticky_expressions:
-            sticky_clause = "is_already_enrolled"
-            if is_desktop:
-                if self.is_rollout:
-                    sticky_clause = "experiment.slug in activeRollouts"
-                else:
-                    sticky_clause = "experiment.slug in activeExperiments"
-            sticky_expressions_joined = " && ".join(
-                [f"({expression})" for expression in sticky_expressions]
+            expressions.append(
+                make_sticky_targeting_expression(
+                    is_desktop, self.is_rollout, sticky_expressions
+                )
             )
-            sticky_expression = f"({sticky_clause}) || ({sticky_expressions_joined})"
-            expressions.append(sticky_expression)
         else:
             expressions.extend(sticky_expressions)
 
         if prefs := self._get_targeting_pref_conflicts():
-            expressions.extend(f"!('{pref}'|preferenceIsUserSet)" for pref in prefs)
+            expressions.extend(
+                make_sticky_targeting_expression(
+                    is_desktop,
+                    self.is_rollout,
+                    (f"!('{pref}'|preferenceIsUserSet)" for pref in prefs),
+                )
+            )
 
         #  If there is no targeting defined all clients should match, so we return "true"
         return (
-            " && ".join([f"({expression})" for expression in expressions])
+            " && ".join(f"({expression})" for expression in expressions)
```

Comment:
This might be causing that test failure?

Reply:
No, thats a generator expression that yields strings. It should behave identitically but be faster.

The problem was:

```python
        if prefs := self._get_targeting_pref_conflicts():
            expressions.extend( # <-- calling extend with str instead of append
                make_sticky_targeting_expression(
```

Reply:
Ah yep 👍